api/v1alpha1/providertemplate_types.go

```diff
@@ -15,24 +15,82 @@
 package v1alpha1
 
 import (
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// ChartAnnotationCAPIVersion is an annotation containing the CAPI exact version in the SemVer format associated with a ProviderTemplate.
+const ChartAnnotationCAPIVersion = "hmc.mirantis.com/capi-version"
+
 // ProviderTemplateSpec defines the desired state of ProviderTemplate
 type ProviderTemplateSpec struct {
-	TemplateSpecCommon `json:",inline"`
+	Helm HelmSpec `json:"helm"`
+	// Compatible CAPI provider version set in the SemVer format.
+	CAPIVersion string `json:"capiVersion,omitempty"`
```

Comment:
At the moment this field is a noop since I don't quite understand in which entity it should be propagated and/or which entity should consume it and what validations should be implemented

Reply:
It should be used to check compatibility of the core CAPI provider (we have a dedicated field in the Release and Management objects alongside HMC itself) and other providers.